### PR TITLE
fix(input-selection): greedy input selector now produces a correct selection under all conditions

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

After the last round of changes to the greedy input selector, some of the property tests were failing when testing some edge cases.

# Proposed Solution

Fix the greedy input selector so it can handle all edge cases correctly.

### CODE REVIEW

**Review by iadmytro — APPROVED** [review]

File: packages/input-selection/test/GreedySelection/util.test.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ describe('splitChange', () => {
       undefined,
       () => 10n,
       () => false,
-      2_000_000n
+      0n
```
Comment (mkazlauskas) [inline packages/input-selection/test/GreedySelection/util.test.ts:19]: Do we need a test where this is not 0?
